### PR TITLE
Use SET IDENTITY_INSERT option to allow explicit id

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @bajtos or @superkhau

### CODE REVIEW

File: lib/mssql.js [hunk 1]
```diff
@@ -414,20 +430,26 @@ MsSQL.prototype.buildInsert = function (model, data) {
...
+
+  if (isIdentity && data[modelPKID] == null) {
     //remove the pkid column if it's in the data, since we're going to insert a
     // new record, not update an existing one.
     delete data[modelPKID];
     //delete the hardcoded id property that jugglindb automatically creates
-    delete data.id;
+    // delete data.id;
   }
-  Object.keys(data).forEach(function (key) {
```
Comment (superkhau) [inline lib/mssql.js]: Is this an optimization for speed?

**Comment by superkhau** [comment]
LGTM

File: lib/mssql.js (hunk 1 above)
Comment (raymondfeng) [inline lib/mssql.js]: Yes.

Comment (superkhau) [inline lib/mssql.js]: Make sense, its actually much faster according to http://jsperf.com/foreach-vs-loop/77

File: lib/mssql.js [hunk 2]
```diff
@@ -155,18 +155,34 @@ MsSQL.prototype.create = function (model, data, callback) {
...
+  sql += "INSERT INTO " + tblName + columns + MsSQL.newline;
   sql += "OUTPUT INSERTED." + this.columnEscaped(model, modelPKID)
     + " AS insertId" + MsSQL.newline;
   sql += (fieldsAndData.paramPlaceholders ?
     "VALUES (" + fieldsAndData.paramPlaceholders + ");" :
     "DEFAULT VALUES;");
 
+  if (isIdentity && data[modelPKID] != null) {
+    sql += "SET IDENTITY_INSERT " + tblName + " OFF;" + MsSQL.newline;
+  }
```
Comment (bajtos) [inline lib/mssql.js]: What happens when the `INSERT` command fails, will MySQL continue executing other commands included in the multi-command string? If it does not, then `IDENTITY_INSERT` won't be turned off when `INSERT` failed.

**Comment by sochka** [comment]
This commit results in the end-users' ability to add arbitrary ids to the model during insertion, which, I believe, should not be possible be default, since this is usually a security problem. And because of this you have to manually validate every single model that user doesn't manually insert any ids.

**Comment by sochka** [comment]
Even more, it's highly possible that many websites using this connector which updated to the new version and didn't notice this change, are now subjects of attacks. If earlier it wasn't possible to post /api/model/ {id: 2147483648}, now - it is and this allows to down the database.

**Comment by sochka** [comment]
adding @superkhau , @raymondfeng for notification.

**Pull request merged**